test/functional/options/defaults_spec.lua

```diff
@@ -0,0 +1,85 @@
+local helpers = require('test.functional.helpers')
+local Screen = require('test.functional.ui.screen')
+local clear, feed = helpers.clear, helpers.feed
+local eval, eq, neq = helpers.eval, helpers.eq, helpers.neq
+local execute, source, expect = helpers.execute, helpers.source, helpers.expect
+
+local function init_session(...)
+  local args = { helpers.nvim_prog, '-i', 'NONE', '--embed',
+    '--cmd', 'set shortmess+=I background=light noswapfile noautoindent',
+    '--cmd', 'set laststatus=1 undodir=. directory=. viewdir=. backupdir=.'
+    }
+  for _, v in ipairs({...}) do
+    table.insert(args, v)
+  end
+  helpers.set_session(helpers.spawn(args))
+end
+
+describe('startup defaults', function()
```

Comment:
These tests protect against regressions for the various cases established in https://github.com/neovim/neovim/pull/4252 , but there's no test for the bug actually fixed by this PR: it would require setting `$HOME` to avoid conflicts with existing vimrc, and lua doesn't make this easy AFAICT.